src/server.cpp

```diff
@@ -127,6 +127,93 @@ void *ServerThread::run()
 	return nullptr;
 }
 
+class MapThread : public Thread
+{
+public:
+
+	MapThread(Server *server):
+		Thread("Server"),
+		m_server(server)
+	{}
+
+	void *run();
+
+private:
+	Server *m_server;
+};
+
+void *MapThread::run()
+{
+	BEGIN_DEBUG_EXCEPTION_HANDLER
+
+	thread_local const float save_interval =
+		g_settings->getFloat("server_map_save_interval") * 1000;
+
+	u64 last_time = 0;
+	while (!stopRequested()) {
+		sleep_ms(100); // no busy loop
+
+		u64 curr_time = porting::getTimeMs();
+		if (curr_time - last_time < save_interval)
+			continue;
+
+		last_time = curr_time;
+
+		ServerEnvironment &env = m_server->getEnv();
+		ServerMap &map = (ServerMap&) env.getMap();
+
+		// Save map, players and auth stuff
+		CompressedBlockList compressed_blocks;
+		SerializedBlockList serialized_blocks;
+		{
+			MutexAutoLock lock(m_server->m_env_mutex);
+
+			// Save ban file
+			if (m_server->getBanManager().isModified()) {
+				m_server->getBanManager().save();
+			}
+
+			// 1. Map: Collect all changed blocks
+			map.save(MOD_STATE_WRITE_NEEDED, true);
+
+			// Save players
+			env.saveLoadedPlayers();
+
+			// Save environment metadata
+			env.saveMeta();
+		} // release the lock here
+
+		infostream << "Collect " << porting::getTimeMs() - curr_time << std::endl;
+
+		static const float map_timer_and_unload_dtime = 2.92;
+		{
+			MutexAutoLock lock(m_server->m_env_mutex);
+			// now run map.timerupdate, collect unloaded blocks
+			map.timerUpdate(map_timer_and_unload_dtime,
```

Comment:
Timer updates are now `server_map_save_interval` dependent. If a server owner chooses a too high value (for whatever reason), the timers will not be executed. Is it necessary to isolate this part from AsyncRunStep?

Reply:
I had internally debated to manage two intervals (it's certainly not hard to do).
In the end, I'd rather document this as map-service-interval (or something), that saves the metadata, data, and performs the timerUpdate, no need to have these on different schedules... I think.

Reply:
Oh and for your question... It's necessary to isolate the block unloading since most of the unloaded blocks will need to be saved (we - unhelpfully - serialized the timestamp into the block's blob, so the entire block needs to be saved again). And getting the save-stalling out of AsyncRunStep is the point of this PR.

If timerUpdate and save-interva really need to be separate I can certainly make it so.